met-api/src/met_api/services/engagement_service.py

```diff
@@ -77,23 +78,95 @@ def publish_scheduled_engagements():
         print('Engagements published: ', engagements)
         return engagements
 
-    def create_engagement(self, data: EngagementSchema):
+    @staticmethod
+    def create_engagement(request_json: dict):
         """Create engagement."""
-        self.validate_fields(data)
-        return EngagementModel.create_engagement(data)
+        # TODO add schema and remove this validation
+        EngagementService.validate_fields(request_json)
+        eng_model = EngagementService._create_engagement_model(request_json)
+        if request_json.get('status_block'):
+            EngagementService._create_eng_status_block(eng_model.id, request_json)
+
+        return eng_model.find_by_id(eng_model.id)
+
+    @staticmethod
+    def _create_engagement_model(engagement_data: dict) -> EngagementModel:
+        """Save engagement."""
+        new_engagement = EngagementModel(
+            name=engagement_data.get('name', None),
+            description=engagement_data.get('description', None),
+            rich_description=engagement_data.get('rich_description', None),
+            start_date=engagement_data.get('start_date', None),
+            end_date=engagement_data.get('end_date', None),
+            status_id=Status.Draft.value,
+            created_by=engagement_data.get('created_by', None),
+            created_date=datetime.utcnow(),
+            updated_by=engagement_data.get('updated_by', None),
+            updated_date=None,
+            published_date=None,
+            scheduled_date=None,
+            banner_filename=engagement_data.get('banner_filename', None),
+            content=engagement_data.get('content', None),
+            rich_content=engagement_data.get('rich_content', None)
+        )
+        new_engagement.save()
+        return new_engagement
 
-    def update_engagement(self, data: EngagementSchema):
+    @staticmethod
+    def _create_eng_status_block(eng_id, engagement_data: dict):
+        """Save engagement."""
+        for status in engagement_data.get('status_block'):
+            new_status_block: EngagementStatusBlockModel = EngagementStatusBlockModel(
+                engagement_id=eng_id,
+                survey_status=status.get('survey_status'),
+                block_text=status.get('block_text')
+            )
+
+            new_status_block.save()
```

Comment:
any chance we can save in bulk rather than save one by one. calling DB on every iteration in a loop is not very performative perhaps.